essential_reading/DataSources/Data_S3.md

```diff
@@ -0,0 +1,236 @@
+[command line interface](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/configure/index.html).
+
+To create a new AWS profile:
+
+```sh
+> aws configure --profile osn-renci
+AWS Access Key ID :
+AWS Secret Access Key :
+Default region name : us-east-1
+Default output format: json
+```
+
+The first two prompts will ask for key/security information you were assigned for access to object storage.
+Answer last questions with "us-east-1" (region) and "json" (format).
+
+Note that this configuration is specific to the OSN 'pod' storage.
+Your profile name and region may be different.
+
+## Endpoints
+
+For storage operations, the AWS system needs the web address of the access point, or _endpoint_
+where it should address filesystem operations. If your storage is completely within the Amazon
+ecosystem, you will likely not need to specify an endpoint.
+However, for 3rd-party storage (such as the OSN pod), you will need to explicitly declare the
+endpoint when the filesystem is first referenced using `fsspec`.
+
+```python
+fs_write = fsspec.filesystem(
+    's3',
+    profile='osn-renci',  ## This is the profile name you configured above.
+    client_kwargs={'endpoint_url': 'https://renc.osn.xsede.org'}
+)
+```
+
+The enpoint URL will be given to you when your key ID and access key are assigned.
+
+## Writing Data to S3
+
+With greater permissions, you may be able to do more destructive activities (overwriting, removing, etc).
+Examples:
+
+* `mkdir` -- makes a new directory / folder
+* `mv` -- moves/renames a file or folder
+* `rm` -- removes a file or folder
+
+From within your Python program, writes to S3 storage can be achieved a few different ways.
+Often, the most convenient is to use a `mapper` to connect a file-like python object to
+the S3 object storage location:
+
+```python
+fname='s3://rsignellbucket2/testing/outfile.zarr'
+outfile=fs_write.get_mapper(fname)
+xarray_dataset.to_zarr(outfile)
+```
+
+The `outfile` variable can be used most anywhere that a file-like object is needed for
+writing.
+
+See the [API documentation](https://filesystem-spec.readthedocs.io/en/latest/api.html)
+for the full details of available operations.
+
+## Read-only access with "requester pays"
+
+Some datasets, such as the first example above (anonymous reading), are offered with 'egress fees' paid by the data owner.
+This means that all access is free to anybody.
+The cost of the network bandwidth used to serve the data is covered by the data's host.
+Not all "public" data is offered this way: it is still available to anybody who wants to read it, but
+the access fees must be paid by the reador (i.e. the 'requester').
+
+When you access a requester-pays dataset, your profile identifies the account which will be
+billed for access.  Open such datasets with an extra option to `fsspec`:
+
+```python
+fs = fsspec.filesystem(
+    's3',
+    profile='osn-renci',
+    anon=False,
+    requester_pays=True
+)
+fs.ls('s3://esip-qhub/noaa/nwm/')
```

Comment:
Suggesting we use an OSN pod endpoint here since the fsspec fs was set up for that above.
```suggestion
fs.ls('usgs-scratch/')
```